presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -234,14 +234,22 @@ private static BingTile getTileCoveringLowerRightCorner(Envelope envelope, int z
         return tile;
     }
 
-    private static void checkGeometryToBingTilesLimits(OGCGeometry geometry, boolean pointOrRectangle, int tileCount)
+    private static void checkGeometryToBingTilesLimits(OGCGeometry geometry, boolean pointOrRectangle, long tileCount)
     {
         if (pointOrRectangle) {
-            checkCondition(tileCount <= 1_000_000, "The number of input tiles is too large (more than 1M) to compute a set of covering bing tiles.");
+            checkCondition(tileCount <= 1_000_000, "The number of input tiles is too large (more than 1M) to compute a set of covering Bing tiles.");
         }
         else {
-            long complexity = ((long) tileCount) * getPointCount(geometry.getEsriGeometry());
-            checkCondition(complexity <= 25_000_000, "The zoom level is too high or the geometry is too complex to compute a set of covering bing tiles. " +
+            checkCondition((int) tileCount == tileCount, "The zoom level is too high to compute a set of covering Bing tiles.");
```

Comment:
You were saying previously that this check is needed to `avoid long overflow when calculating complexity`. Now that you handle the overflow case by catching `ArithmeticException` below, do you still need this?

Reply:
@nezihyigitbasi I want to keep this check so I can provide a more specific error message - "zoom level is too high" - as opposed to more generic one - "zoom level is too high or the geometry is too complex".